crates/bevy_asset/src/io/metadata.rs

```diff
@@ -0,0 +1,58 @@
+/// A enum representing a type of file.
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+pub enum FileType {
```

Comment:
```suggestion
#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
#[non_exhaustive]
pub enum FileType {
```

It might be worth adding the `non_exhaustive` attribute in case we want to add additional types in the future? This would prevent large breaks for existing code.